bio-qiime-pipeline/src/import_multiplexed.py

```diff
@@ -0,0 +1,132 @@
+from typing import Optional, List
+from pathlib import Path
+from zipfile import ZipFile
+
+import logging
+
+from coretex import CustomDataset, CustomSample, Experiment, folder_manager
+from coretex.bioinformatics import ctx_qiime2
+
+from .utils import convertMetadata
+from .caching import getCacheNameOne
+
+
+FORWARD_FASTQ = "forward.fastq"
+REVERSE_FASTQ = "reverse.fastq"
+BARCODES_FASTQ = "barcodes.fastq"
+
+
+def importSample(sequencesPath: Path, sequenceType: str, outputDir: Path) -> Path:
+    importedSequencesPath = outputDir / "multiplexed-sequences.qza"
+
+    ctx_qiime2.toolsImport(sequenceType, str(sequencesPath), str(importedSequencesPath))
+
+    outputPath = outputDir / "import-output.zip"
+    with ZipFile(outputPath, "w") as outputFile:
+        outputFile.write(importedSequencesPath, importedSequencesPath.name)
+
+    return outputPath
+
+
+def importMetadata(metadataPath: Path, outputDir: Path) -> Path:
+    metadataPath = convertMetadata(metadataPath)
+
+    outputPath = outputDir / "metadata-output.zip"
+    with ZipFile(outputPath, "w") as outputFile:
+        outputFile.write(metadataPath, metadataPath.name)
+
+    return outputPath
+
+
+def getFastq(sample: CustomSample, fileName: str) -> Optional[Path]:
+    foundFiles = list(sample.path.glob(f"*{fileName}"))
+    foundFiles.extend(list(sample.path.glob(f"*{fileName}.gz")))
+
+    if len(foundFiles) > 1:
+        raise RuntimeError(f">> [Qiime: Import] Found multiple {fileName} files")
+
+    return foundFiles[0] if len(foundFiles) == 1 else None
+
+
+def prepareSequences(
+    barcodesPath: Path,
+    forwardPath: Path,
+    reversePath: Optional[Path]
+) -> Path:
+
+    sequenceFolderPath = folder_manager.createTempFolder("sequencesFolder")
+
+    newBarcodesPath = sequenceFolderPath / f"{BARCODES_FASTQ}.gz"
+    newForwardPath = sequenceFolderPath / f"{FORWARD_FASTQ}.gz"
+    newReversePath = sequenceFolderPath / f"{REVERSE_FASTQ}.gz"
+
+    if reversePath is None:
+        checkGz = [path.suffix == ".gz" for path in [barcodesPath, forwardPath]]
+    else:
+        checkGz = [path.suffix == ".gz" for path in [barcodesPath, forwardPath, reversePath]]
+
+    if all(checkGz):
```

Comment:
allow mix of both, update `getFastq` function to compress the fastq file if it is not already compressed with gz